ctru-rs/src/linear.rs

```diff
@@ -47,3 +51,32 @@ unsafe impl Allocator for LinearAllocator {
         }
     }
 }
+
+/// Trait indicating a type has been allocated using [`LinearAllocator`].
+/// This can be used to enforce that a given slice was allocated in LINEAR memory.
+#[diagnostic::on_unimplemented(
+    message = "{Self} is not allocated with `ctru::linear::LinearAllocator`"
+)]
+pub trait LinearAllocation: Sealed {}
```

Comment:
As an alternative to sealing this trait, we could make it `unsafe` to implement, which would allow users to define it on their own types (while being forced to acknowledge the risk if their type is not _actually_ allocated with LINEAR). The alternative is to just force them to convert to `Vec` etc.

Reply:
I heavily suggest unsealing the implementation of the `LinearAllocation`, since it would massively simplify interoperability with external types and libraries. Less boilerplate conversion code is a great thing and the responsibility for the `unsafe` implementation would fall on the user anyways.